libs/sync/sync.go

```diff
@@ -132,15 +154,44 @@ func (s *Sync) notifyComplete(ctx context.Context, d diff) {
 }
 
 func (s *Sync) RunOnce(ctx context.Context) error {
-	// tradeoff: doing portable monitoring only due to macOS max descriptor manual ulimit setting requirement
-	// https://github.com/gorakhargosh/watchdog/blob/master/src/watchdog/observers/kqueue.py#L394-L418
-	all, err := s.fileSet.All()
+	all := make([]fileset.File, 0)
+	if s.SyncOptions.Include == nil {
+		// tradeoff: doing portable monitoring only due to macOS max descriptor manual ulimit setting requirement
+		// https://github.com/gorakhargosh/watchdog/blob/master/src/watchdog/observers/kqueue.py#L394-L418
+		gitFiles, err := s.fileSet.All()
+		if err != nil {
+			log.Errorf(ctx, "cannot list files: %s", err)
+			return err
+		}
+		all = append(all, gitFiles...)
```

Comment:
We should always synchronize the Git fileset. Users can use includes to add files that are ignored by Git, and excludes to mask files out that are observed by Git but should not be synced.